File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/java/shortest-paths.java
/**
 * Java Implementation: Shortest Paths (Dijkstra, Bellman-Ford, Floyd-Warshall)
 * Invariant: Dijkstra O((V+E) log V), Floyd O(V^3)
 */
package seed.dsa.core;

import java.util.*;

public class ShortestPathsDijkstraBellmanFordFloydWarshall<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}

File: src/courses/data/catalog/02-dsa/advanced-dsa/implementations/csharp/shortest-paths.cs
/**
 * C# Implementation: Shortest Paths (Dijkstra, Bellman-Ford, Floyd-Warshall)
 * Invariant: Dijkstra O((V+E) log V), Floyd O(V^3)
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class ShortestPathsDijkstraBellmanFordFloydWarshall<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}
